google/cloud/bigtable/mutations_batcher.py

```diff
@@ -15,16 +15,140 @@
 from __future__ import annotations
 
 import asyncio
-from typing import TYPE_CHECKING
+import atexit
+import warnings
+from typing import Any, Awaitable, TYPE_CHECKING
 
+from google.cloud.bigtable.mutations import RowMutationEntry
+from google.cloud.bigtable.exceptions import MutationsExceptionGroup
+from google.cloud.bigtable.exceptions import FailedMutationEntryError
+
+from google.cloud.bigtable._mutate_rows import _MutateRowsOperation
+from google.cloud.bigtable._mutate_rows import MUTATE_ROWS_REQUEST_MUTATION_LIMIT
 from google.cloud.bigtable.mutations import Mutation
-from google.cloud.bigtable.row_filters import RowFilter
 
 if TYPE_CHECKING:
     from google.cloud.bigtable.client import Table  # pragma: no cover
 
-# Type alias used internally for readability.
-_row_key_type = bytes
+# used to make more readable default values
+MB_SIZE = 1024 * 1024
+
+
+class _FlowControl:
+    """
+    Manages flow control for batched mutations. Mutations are registered against
+    the FlowControl object before being sent, which will block if size or count
+    limits have reached capacity. As mutations completed, they are removed from
+    the FlowControl object, which will notify any blocked requests that there
+    is additional capacity.
+
+    Flow limits are not hard limits. If a single mutation exceeds the configured
+    limits, it will be allowed as a single batch when the capacity is available.
+    """
+
+    def __init__(
+        self,
+        max_mutation_count: int | None,
+        max_mutation_bytes: int | None,
+    ):
```

Comment:
I thought we were dropping None?

Reply:
I think I was still unsure whether we wanted to fully remove the behaviour of `max... = None` (by adding a hard cap), or just discourage it by removing None from the signature.

I dropped it from the signature for now, but let me know if you think we need to add a cap too